miden/tests/integration/stdlib/math/secp256k1_base_field.rs

```diff
@@ -0,0 +1,562 @@
+        borrow = v.0;
+        b[3] = v.1;
+
+        let v = Self::sbb(prime[4], self.limbs[4], borrow);
+        borrow = v.0;
+        b[4] = v.1;
+
+        let v = Self::sbb(prime[5], self.limbs[5], borrow);
+        borrow = v.0;
+        b[5] = v.1;
+
+        let v = Self::sbb(prime[6], self.limbs[6], borrow);
+        borrow = v.0;
+        b[6] = v.1;
+
+        let v = Self::sbb(prime[7], self.limbs[7], borrow);
+        b[7] = v.1;
+
+        Self::Output { limbs: b }
+    }
+}
+
+impl Sub for BaseField {
+    type Output = BaseField;
+
+    /// Computes modular subtraction of one secp256k1 base field element, from another one, in Montgomery form
+    ///
+    /// See https://github.com/itzmeanjan/secp256k1/blob/6e5e654823a073add7d62b21ed88e9de9bb06869/field/base_field.py#L98-L102
+    fn sub(self, rhs: Self) -> Self::Output {
+        self + (-rhs)
+    }
+}
+
+impl PartialEq for BaseField {
+    /// Checks whether two secp256k1 field elements are equal or not, in Montogomery form
+    fn eq(&self, other: &Self) -> bool {
+        let mut flg = false;
+
+        for i in 0..8 {
+            flg |= (self.limbs[i] ^ other.limbs[i]) != 0;
+        }
+
+        !flg
+    }
+
+    /// Checks whether two secp256k1 field elements are not equal to each other, in Montogomery form
+    fn ne(&self, other: &Self) -> bool {
+        !(self == other)
+    }
+}
+
+#[test]
+fn test_montgomery_repr() {
+    let source = "
+    use.std::math::secp256k1_base_field
+
+    begin
+        exec.secp256k1_base_field::to_mont
+        exec.secp256k1_base_field::from_mont
+    end";
+
+    let mut num = [0u32; 8];
+    for i in 0..4 {
+        let a = rand_utils::rand_value::<u32>();
+        let b = rand_utils::rand_value::<u32>();
+
+        num[i] = a;
+        num[i ^ 4] = b;
+    }
+
+    let mut stack = [0u64; 8];
+    for i in 0..4 {
+        stack[i] = num[i] as u64;
+        stack[i ^ 4] = num[i ^ 4] as u64;
+    }
+    stack.reverse();
+
+    let test = build_test!(source, &stack);
+    let strace = test.get_last_stack_state();
+
+    for i in 0..8 {
+        assert_eq!(Felt::new(num[i] as u64), strace[i]);
+    }
+}
+
+#[test]
+fn test_mul() {
+    let source = "
+    use.std::math::secp256k1_base_field
+
+    begin
+        exec.secp256k1_base_field::mul
+    end";
+
+    let mut stack = [0u64; 16];
+
+    let mut elm0 = BaseField { limbs: [0u32; 8] };
+    let mut elm1 = BaseField { limbs: [0u32; 8] };
+
+    for i in 0..8 {
+        let a = rand_utils::rand_value::<u32>();
+        let b = rand_utils::rand_value::<u32>();
+
+        stack[i] = a as u64;
+        stack[i ^ 8] = b as u64;
+
+        elm0.limbs[i] = a;
+        elm1.limbs[i] = b;
+    }
+
+    let elm2 = elm0 * elm1;
+    stack.reverse();
+
+    let test = build_test!(source, &stack);
+    let strace = test.get_last_stack_state();
+
+    for i in 0..8 {
+        assert_eq!(Felt::new(elm2.limbs[i] as u64), strace[i]);
+    }
```

Comment:
1. similar comments to the previous test for this and other tests in this file regarding adjustments for clarity & consistency with other tests
2. It looks like the setup code is duplicated for multiple tests in this file. Could we use a helper function for the setup of the stack and elements instead?